tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php

```diff
@@ -108,4 +108,39 @@ public function shouldThrowExceptionWhenUserOfUserMergeDiffersFromSorting(): voi
         );
         $builder->build();
     }
+
+    /** @test */
+    public function shouldCheckSequenceOfUserIdsWhenMerging(): void
+    {
+        /**
+         * Should pass: ([interaction], [user merge], [recommendation]):
+         *              (A, A -> B, B)
+         */
+        $interactionCommand = Interaction::purchase('old-user-id', 'test-item-id');
+        $userMergeCommand = UserMerge::mergeFromSourceToTargetUser('old-user-id', 'new-user-id');
+        $recommendationsCommand = UserRecommendation::create('new-user-id', 5, 'scenario', 0.5, 3600);
+
+        $builder = new RecommendationRequestBuilder($recommendationsCommand);
+        $builder->setUserMerge($userMergeCommand);
+        $builder->setInteraction($interactionCommand);
+        $this->assertInstanceOf(Request::class, $builder->build());
+
+        /**
+         * Should fail: ([interaction], [user merge], [recommendation]):
+         *              (A, B -> A, A)
+         */
+        $interactionCommand = Interaction::purchase('old-user-id', 'test-item-id');
+        $userMergeCommand = UserMerge::mergeInto('old-user-id', 'new-user-id');
+        $recommendationsCommand = UserRecommendation::create('old-user-id', 5, 'scenario', 0.5, 3600);
+
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage(
+            'User in interaction ("old-user-id") must be the same as user in UserMerge::from ("new-user-id")'
```

Comment:
Isn't the `UserMerge::from` confusing? Would something like `User in interaction ("old-user-id") must be the same as the source user of UserMerge ("new-user-id")` better?